care/facility/migrations/0398_alter_patientconsultation_discharge_reason.py

```diff
@@ -0,0 +1,47 @@
+# Generated by Django 4.2.6 on 2023-12-07 10:52
+
+from django.db import migrations, models
+from django.db.models import Case, Value, When
+
+
+def migrate_discharge_reasons(apps, schema_editor):
+    PatientConsultation = apps.get_model("facility", "PatientConsultation")
+    mapping = {
+        "REC": 1,
+        "REF": 2,
+        "EXP": 3,
+        "LAMA": 4,
+        "": -1,
+    }
+    cases = [When(discharge_reason=k, then=Value(v)) for k, v in mapping.items()]
+    PatientConsultation.objects.filter(discharge_reason__isnull=False).update(
+        discharge_reason=Case(*cases, default=Value(-1))
+    )
+
+
+class Migration(migrations.Migration):
+    dependencies = [
+        ("facility", "0397_truncate_discharge_time"),
+    ]
+
+    operations = [
+        migrations.RunPython(
+            migrate_discharge_reasons, reverse_code=migrations.RunPython.noop
```

Comment:
We are migrating all arbitrary values for discharge reason to -1, how do we handle that when reversing? 😅

Reply:
> If we apply this, reverting to the exact state as before will not be possible, but if we revert and apply the migration again we will lose all data

As a safeguard from that situation, should we skip the migration code in case the discharge_reason for any record is an integer? Sounds hacky though :/

Reply:
or we could remove reverse migration, to avoid losing data

Reply:
@rithviknishad which approach do you suggest?

Reply:
The usual practice i implement in all standard architectures is that you never ever have migrations that changes fields ( table schema ) and code that uses that in one single change,
we usually introduce the fields ( table structure changes  ) in an MR, this is just the migrations that add the required fields. if the field is being updated, you create a temporary field. 
once the migration with the change is implemented, you go ahead and start making code that uses that particular field as a separate MR, this would contain all the migrations that migrate the actual data.  this MR would also contain code that starts to use the temp field
Once all the migrations have been performed and the system is running fine, another MR is made to move everything back to the way it was and deprecate the field ( note: deprecate not delete )

This way if any of the deployments fail, the application continues to work as intended with minimal downtime. each individual migrations would also contain reverse migrations that undo just their task.

Reply:
I understand, I've converted this PR to just the schema change. Data migration along with the reverse code + code changes to use the new field will be in the next PR, then after that moving back the new field to the old field in another PR.